classes/mail/variables/SubmissionEmailVariable.inc.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * @file classes/mail/variables/SubmissionEmailVariable.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionEmailVariable
+ * @ingroup mail_variables
+ *
+ * @brief Represents variables associated with a submission that can be assigned to a template
+ */
+
+namespace PKP\mail\variables;
+
+use APP\article\Author;
+use PKP\core\PKPApplication;
+use APP\facades\Repo;
+use PKP\i18n\PKPLocale;
+use PKP\publication\PKPPublication;
+use PKP\submission\PKPSubmission;
+
+class SubmissionEmailVariable extends Variable
+{
+    const SUBMISSION_TITLE = 'submissionTitle';
+    const SUBMISSION_ID = 'submissionId';
+    const SUBMISSION_ABSTRACT = 'submissionAbstract';
+    const AUTHORS = 'authors';
+    const AUTHORS_FULL = 'authorsFull';
+    const SUBMISSION_URL = 'submissionUrl';
+
+    protected PKPSubmission $submission;
+
+    protected PKPPublication $currentPublication;
+
+    /**
+     * @param PKPSubmission $submission
+     */
+    public function __construct(PKPSubmission $submission)
+    {
+        $this->submission = $submission;
+        $currentPublicationId = $this->submission->getData('currentPublicationId');
+        $this->currentPublication = Repo::publication()->get($currentPublicationId);
```

Comment:
This method may return null but according to my tests `currentPublicationId` is always available. Is it a mistake?

Reply:
It's technically possible for a submission to exist without a publication, though I believe it will blow a lot of things up! :laughing: 

If a publication exists, it will be available at `$this->submission->getCurrentPublication()`. If it doesn't, going to the `Repo` won't help. It will still return `null`.

Reply:
Thanks, fixed.